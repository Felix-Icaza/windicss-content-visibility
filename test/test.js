import { Processor } from 'windicss/lib'
import contentVisibility from '../src/index'

describe('Content Visibility Plugin', () => {
  it('should generate content-visibility properties correctly', () => {
    const processor = new Processor({
      plugins: [contentVisibility]
    })

    const css = `
      content-auto
      content-visible
      content-hidden
    `
    const result = processor.interpret(css)

    expect(result.ignored.length).toEqual(0)
    expect(result.styleSheet.build()).toMatchSnapshot('content-visibility')
  })

  it('generate contain-intrinsic-size correctly', () => {
    const processor = new Processor({
      plugins: [contentVisibility]
    })

    const css = `
      intrinsic-size-200
      intrinsic-size-350
      intrinsic-size-525
      intrinsic-size-1200
      intrinsic-size-9999
    `
    const result = processor.interpret(css)

    expect(result.ignored.length).toEqual(0)
    expect(result.styleSheet.build()).toMatchSnapshot('contain-intrinsic-size')
  })

  it('check responsive variants', () => {
    const processor = new Processor({
      plugins: [contentVisibility]
    })

    const css = `
      sm:content-auto
      md:content-visible
      lg:content-hidden
      xl:content-auto
      sm:intrinsic-size-200
      md:intrinsic-size-350
      lg:intrinsic-size-525
      xl:intrinsic-size-700
    `
    const result = processor.interpret(css)

    expect(result.ignored.length).toEqual(0)
    expect(result.styleSheet.build()).toMatchSnapshot('responsive variants')
  })
})
