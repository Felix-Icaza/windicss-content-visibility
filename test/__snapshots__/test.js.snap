// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Visibility Plugin check responsive variants: responsive variants 1`] = `
"@media (min-width: 640px) {
  .sm\\\\:content-auto {
    content-visibility: auto;
  }
  .sm\\\\:intrinsic-size-200 {
    contain-intrinsic-size: 200px;
  }
}
@media (min-width: 768px) {
  .md\\\\:content-visible {
    content-visibility: visible;
  }
  .md\\\\:intrinsic-size-350 {
    contain-intrinsic-size: 350px;
  }
}
@media (min-width: 1024px) {
  .lg\\\\:content-hidden {
    content-visibility: hidden;
  }
  .lg\\\\:intrinsic-size-525 {
    contain-intrinsic-size: 525px;
  }
}
@media (min-width: 1280px) {
  .xl\\\\:content-auto {
    content-visibility: auto;
  }
  .xl\\\\:intrinsic-size-700 {
    contain-intrinsic-size: 700px;
  }
}"
`;

exports[`Content Visibility Plugin generate contain-intrinsic-size correctly: contain-intrinsic-size 1`] = `
".intrinsic-size-200 {
  contain-intrinsic-size: 200px;
}
.intrinsic-size-350 {
  contain-intrinsic-size: 350px;
}
.intrinsic-size-525 {
  contain-intrinsic-size: 525px;
}
.intrinsic-size-1200 {
  contain-intrinsic-size: 1200px;
}
.intrinsic-size-9999 {
  contain-intrinsic-size: 9999px;
}"
`;

exports[`Content Visibility Plugin should generate content-visibility properties correctly: content-visibility 1`] = `
".content-auto {
  content-visibility: auto;
}
.content-visible {
  content-visibility: visible;
}
.content-hidden {
  content-visibility: hidden;
}"
`;
